src/object.rs

```diff
@@ -680,6 +690,40 @@ impl Object {
     unsafe { v8__Object__SetAlignedPointerInInternalField(self, index, value) }
   }
 
+  /// Wraps a JS wrapper with a C++ instance.
+  #[allow(clippy::not_unsafe_ptr_arg_deref)]
+  #[inline(always)]
+  pub fn wrap(
```

Comment:
Please consider marking this function as `unsafe` so that `deno_core` may assume that no one else calls it. `#[allow(clippy::not_unsafe_ptr_arg_deref)]` can be removed.

Reply:
there's nothing unsafe about this function. its only unsafe to dereference the pointer you get from `unwrap`, but that risk is taken by `try_unwrap_cppgc_object` in `deno_core`, not these functions.

Reply:
Should the `#[cppgc]` attribute be considered unsafe to use then? Changing every instance of `#[cppgc]` in parameter position to something like `#[unsafe(cppgc)]` sounds like much more work.

Reply:
> there's nothing unsafe about this function.

On second thought I agree about this. Anyway something should be done to ensure that a user of `deno_core` cannot cause UB without writing `unsafe`.

Reply:
I think on the `deno_core` side it can be safe if we restructure it slightly:

for each cppgc type: generate a functiontemplate (required anyway) and unique tag. when wrapping and unwrapping the tag is used. if the tag mismatches, the unwrap will fail. this makes ops safe so we don't need `#[unsafe(cppgc)]`.

if an embedder goes around this and uses `Object::unwrap` directly they must use `unsafe` to dereference the pointer it returns, so they accept liability in that case.

Reply:
There are only about 30000 possible tags making it too easy to guess the tag of a specific type.

Just found out that apparently V8 [dereferences the pointer during tracing if it's non-null](https://source.chromium.org/chromium/chromium/src/+/main:v8/src/heap/marking-visitor-inl.h;l=236-240;drc=a47eabc951c43363341082dddd54684f568fcf58) under the assumption that it's pointing to memory allocated by `cppgc::MakeGarbageCollected` (it expects a pointer-sized header located right before the pointed object). Excuse me for being so inconsistent, but for this reason I think `wrap` is definitely `unsafe`.